### PR TITLE
Add endpoint to list system labels

### LINKED ISSUE #4492: Hide Kubermatic-specific labels from the user
**User Story** https://github.com/kubermatic/dashboard-v2/issues/1675 will block adding Kubermatic-specific labels to the system. We can also hide them completely and not return them from our API as this is an implementation detail.

cc @alvaroaleman

### PR DESCRIPTION
**Pull request by floreks**
**What this PR does / why we need it**:
Add endpoint to list restricted system labels and filters them out from Cluster, ND, Project LIST/GET endpoints.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4492

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by maciaszczykm** [comment]
LGTM, leaving for @zreigz.

**Comment by zreigz** [comment]
It would be good to add some unit tests for this new endpoint

File: api/pkg/handler/v1/label/label_test.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	t.Parallel()
+	testcases := []struct {
+		Name            string
+		ExpectedLabels  apiv1.ResourceLabelMap
+		HTTPStatus      int
+		ExistingAPIUser *apiv1.User
+	}{
+		// scenario 1
+		{
+			Name:            "scenario 1: list clusters that belong to the given project",
```
Comment (zreigz) [inline api/pkg/handler/v1/label/label_test.go]: change description

File: api/pkg/handler/routes_v1.go [hunk 2]
```diff
@@ -527,6 +528,10 @@ func (r Routing) RegisterV1(mux *mux.Router, metrics common.ServerMetrics) {
 	mux.Methods(http.MethodGet).
 		Path("/me").
 		Handler(r.getCurrentUser())
+
+	mux.Methods(http.MethodGet).
+		Path("/label/system").
```
Comment (zreigz) [inline api/pkg/handler/routes_v1.go]: we use the plural for the endpoints. should be `/labels/system`

**Review by zreigz — CHANGES_REQUESTED** [review]

**Pull request merged**